### ISSUE TITLE
DisallowImplicitArrayCreation false positive

### ISSUE THREAD

**Issue by dereuromark** [slevomat/coding-standard#516]
Our codebase contains a piece:
```php
// class Container implements ArrayAccess
    /**
     * @param string $key
     * @param mixed $value
     *
     * @return $this
     */
    public function set($key, $value)
    {
        $this[$key] = $value;

        return $this;
    }
```
Resulting in
```
 46 | ERROR | Implicit array creation is disallowed.
```

I think this specific case should be exluded as it is valid php code afaik.

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/160ab10d367aec8ac5e073c5e258a12b67fedb40
